src/service/performance-impl.js

```diff
@@ -94,15 +94,13 @@ export class Performance {
     this.isMessagingReady_ = false;
 
     /** @private @const {!Promise} */
-    this.whenReadyToRetrieveResourcesPromise_ = new Promise(resolve => {
-      onDocumentReady(this.win.document, () => {
-        // We need to add a delay, since this can execute earlier
-        // than the onReady callback registered inside of `Resources`.
-        // Should definitely think of making `getResourcesInViewport` async.
-        timer.delay(resolve);
-      });
-    });
-
+    this.whenReadyToRetrieveResourcesPromise_ =
+        whenDocumentReady(this.win.document)
+        .then(() => {
+          // Two fold. First, resolve the promise to undefined.
+          // Second, causes a delay until the promise resolves so that
```

Comment:
Maybe add a bit more explanation of why this causes a delay. It may look like a no-op to folks not very familiar with how promises work.